packages/bundler/src/modules/MempoolManager.ts

```diff
@@ -290,4 +290,8 @@ export class MempoolManager {
 
     return res.filter(it => it != null).map(it => (it as string).toLowerCase())
   }
+
+  getMempool (): MempoolEntry[] {
```

Comment:
I am concerned that the `removeUserOp` function is only triggered when bundler attempts to include a transaction and it fails. Before this PR the `mempool` was basically private so this did not matter. But now, you rely on the mempool to only have valid Paymaster transactions to decide if you are adding a new UserOp or not. 
Practically, your mempool may be choke-full of invalid Paymaster transactions that it never gets to do a second validation for, and therefore refuses to add valid UserOps for this Paymaster instead.

Reply:
its a separate bug: removeUserOp should be triggered only by events, and not after submitting a bundle (should be opened as a separate bug, unrelated to this erep)